service/main.py

```diff
@@ -5,34 +5,84 @@
 import aiogram.exceptions
 from aiogram import Bot, Dispatcher
 from aiogram.filters import CommandStart
-from aiogram.types import Message
+from aiogram.handlers import MessageHandler, CallbackQueryHandler
+from aiogram.types import InlineKeyboardButton
+from aiogram.filters.callback_data import CallbackData
+from aiogram.utils.keyboard import InlineKeyboardBuilder
+from sqlalchemy import select
 from sqlalchemy_helpers.aio import get_or_create
 
 import settings
 from database import async_session
-from database.models import User
+from database.models import User, Course
 
 logging.config.dictConfig(settings.LOGGING_CONFIG)
 logger = logging.getLogger(__name__)
 
 dispatcher = Dispatcher()
 
 
+class CourseCallbackData(CallbackData, prefix="course"):
+    code: str
+
+
 @dispatcher.message(CommandStart())
-async def process_start_command(message: Message):
-    async with async_session() as session:
-        await get_or_create(
-            session,
-            User,
-            tg_id=message.from_user.id,
-            first_name=message.from_user.first_name,
-            last_name=message.from_user.last_name,
-            username=message.from_user.username,
-            language_code=message.from_user.language_code
-        )
-        await session.commit()
-
-    await message.answer('Hi!')
+class StartCommandHandler(MessageHandler):
+    """
+    Handler of the start command.
+
+    The start command takes or creates a user and displays a list of 
+    available courses for him to start.
+    """
+
+    async def handle(self):
+        async with async_session() as session:
+            await get_or_create(
+                session,
+                User,
+                tg_id=self.from_user.id,
+                first_name=self.from_user.first_name,
+                last_name=self.from_user.last_name,
+                username=self.from_user.username,
+                language_code=self.from_user.language_code
+            )
+            await session.commit()
+
+            courses = await session.scalars(select(Course))
+
+        await self.answer_available_courses(courses.all())
+
+    async def answer_available_courses(self, courses: list[Course]):
+        for course in courses:
+            builder = InlineKeyboardBuilder()
```

Comment:
You can use InlineKeyboardMarkup instead of InlineKeyboardBuilder.

Reply:
Fixed